test/json/skip_suite.cpp

```diff
@@ -0,0 +1,180 @@
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.symbol(), token::symbol::boolean);
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.value<bool>(), true);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "true");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::string);
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.value<std::string>(), "foo");
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "\"foo\"");
+}
+
+void test_array()
+{
+    const char input[] = R"([
+        [],
+        42,
+        [ false ],
+        43,
+        [ "test", false, 33 ],
+        44,
+        [ "test", false, [32, "foo"] ],
+        45,
+        [ "test", false, {"bar": "bar"} ],
+        46
+    ])";
+    json::reader reader(input);
+    json::reader::view_type skipped;
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::begin_array);
+    TRIAL_PROTOCOL_TEST(reader.next());
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::begin_array);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "[]");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::integer);
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.value<int>(), 42);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "42");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::begin_array);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "[ false ]");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::integer);
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.value<int>(), 43);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "43");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::begin_array);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, R"([ "test", false, 33 ])");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::integer);
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.value<int>(), 44);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "44");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::begin_array);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, R"([ "test", false, [32, "foo"] ])");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::integer);
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.value<int>(), 45);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "45");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::begin_array);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, R"([ "test", false, {"bar": "bar"} ])");
+
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.code(), token::code::integer);
+    TRIAL_PROTOCOL_TEST_EQUAL(reader.value<int>(), 46);
+    skipped = json::partial::skip(reader);
+    TRIAL_PROTOCOL_TEST_EQUAL(skipped, "46");
+}
+
+void test_errored()
```

Comment:
I would break this into one test case per error.

I generally call such test cases names staring with `fail_` instead of `test_`